datasets/poem_sentiment/poem_sentiment.py

```diff
@@ -0,0 +1,85 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Poem Sentiment: A sentiment dataset of poem verses"""
+
+from __future__ import absolute_import, division, print_function
+
+import datasets
+
+
+_CITATION = """\
+@misc{sheng2020investigating,
+      title={Investigating Societal Biases in a Poetry Composition System},
+      author={Emily Sheng and David Uthus},
+      year={2020},
+      eprint={2011.02686},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}
+"""
+
+_DESCRIPTION = """\
+Poem Sentiment is a sentiment dataset of poem verses from Project Gutenberg. \
+This dataset can be used for tasks such as sentiment classification or style transfer for poems.
+"""
+
+
+_HOMEPAGE = "https://github.com/google-research-datasets/poem-sentiment"
+
+_BASE_URL = "https://raw.githubusercontent.com/google-research-datasets/poem-sentiment/master/data/"
+_URLS = {
+    "train": f"{_BASE_URL}/train.tsv",
+    "dev": f"{_BASE_URL}/dev.tsv",
+    "test": f"{_BASE_URL}/test.tsv",
+}
+
+
+class PoemSentiment(datasets.GeneratorBasedBuilder):
+    """Poem Sentiment: A sentiment dataset of poem verses"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("int32"),
+                    "verse_text": datasets.Value("string"),
+                    "label": datasets.ClassLabel(names=["negative", "positive", "no_impact"]),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        downloaded_files = dl_manager.download(_URLS)
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["train"]}),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": downloaded_files["dev"]}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": downloaded_files["test"]}),
+        ]
+
+    def _generate_examples(self, filepath):
+        with open(filepath, encoding="utf-8") as f:
+            lines = f.readlines()
+            for line in lines:
+                fields = line.strip().split("\t")
+                idx, verse_text, label = fields
+                label = int(label)
+                label = label + 1 if label == 1 or label == -1 else label
```

Comment:
is it possible to end up with label=1 for positive this way ??
label = -1 -> 0
label = 0 -> 0
label = 1 -> 2

Reply:
My bad. And thanks Quentin, while fixing this, I also realised that there is another label "mixed" which I missed to include